group_project_v2/mixins.py

```diff
@@ -160,22 +327,18 @@ def workgroup(self):
         workgroup = self._get_workgroup(self.project_api, self.user_id, self.course_id)
         return workgroup if workgroup else self.FALLBACK_WORKGROUP
 
-    @staticmethod
-    @memoize_with_expiration()
-    def _get_workgroup(project_api, user_id, course_id):
+    def _get_workgroup(self, project_api, user_id, course_id):
         """
         :rtype: WorkgroupDetails
```

Comment:
@jbzdak as far as I can tell `get_user_roles_for_course` i not memoized in master and not modified here, so it's not actually the same. It won't cause trouble to memoize it though.

Actually, the whole point of this method was to memoize the workgroup (and it was one of the first to be memoized), but since you've removed it it might make sense to inline it - it's not used anywhere except `workgroup` property just above it.

Less radical change would be to drop `project_api` from the list of parameters - it is available as `self.project_api`